sources/procsys.go

```diff
@@ -0,0 +1,210 @@
+// (C) Copyright 2017 Hewlett Packard Enterprise Development LP
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package sources
+
+import (
+	"io/ioutil"
+	"path/filepath"
+	"strconv"
+	"strings"
+
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+const (
+	lnetAllocatedHelp     string = "Number of messages currently allocated"
+	lnetMaximumHelp       string = "Maximum number of outstanding messages"
+	lnetErrorsHelp        string = "Total number of errors"
+	lnetSendCountHelp     string = "Total number of messages that have been sent"
+	lnetReceiveCountHelp  string = "Total number of messages that have been received"
+	lnetRouteCountHelp    string = "Total number of messages that have been routed"
+	lnetDropCountHelp     string = "Total number of messages that have been dropped"
+	lnetSendLengthHelp    string = "Total number of bytes sent"
+	lnetReceiveLengthHelp string = "Total number of bytes received"
+	lnetRouteLengthHelp   string = "Total number of bytes for routed messages"
+	lnetDropLengthHelp    string = "Total number of bytes that have been dropped"
+)
+
+func init() {
+	Factories["procsys"] = newLustreProcSysSource
+}
+
+type lustreProcSysSource struct {
+	lustreProcMetrics []lustreProcMetric
+	basePath          string
+}
+
+func (s *lustreProcSysSource) generateLNETTemplates() error {
+	metricMap := map[string][]lustreHelpStruct{
+		"lnet": {
+			{"catastrophe", "catastrophe_enabled", "Returns 1 if currently in catastrophe mode", s.gaugeMetric},
+			{"console_backoff", "console_backoff_enabled", "Returns non-zero number if console_backoff is enabled", s.gaugeMetric},
+			{"console_max_delay_centisecs", "console_max_delay_centiseconds", "Minimum time in centiseconds before the console logs a message", s.gaugeMetric},
+			{"console_min_delay_centisecs", "console_min_delay_centiseconds", "Maximum time in centiseconds before the console logs a message", s.gaugeMetric},
+			{"console_ratelimit", "console_ratelimit_enabled", "Returns 1 if the console message rate limiting is enabled", s.gaugeMetric},
+			{"debug_mb", "debug_megabytes", "Maximum buffer size in megabytes for the LNET debug messages", s.gaugeMetric},
+			{"fail_err", "fail_err", "Number of errors that have been thrown", s.counterMetric},
```

Comment:
Decided with `fail_error_total` as it seemed to be the most accurate... though I agree about these metrics being weird.